src/ResourceManager/Version2016_09_01/AzureRMCmdlet.cs

```diff
@@ -546,5 +592,27 @@ private void EnqueueDebugSender(object sender, StreamEventArgs args)
         {
             DebugMessages.Enqueue(args.Message);
         }
+
+        public object GetDynamicParameters()
+        {
+            var parameters = new RuntimeDefinedParameterDictionary();
+
+            // add `-SubscriptionId` if the cmdlet has [SupportsSubscriptionId] attribute
+            if (GetType().IsDefined(typeof(SupportsSubscriptionIdAttribute), true))
+            {
+                const string helpMessage = @"The ID of the subscription.
+By default, cmdlets are executed in the subscription that is set in the current context. If the user specifies another subscription, the current cmdlet is executed in the subscription specified by the user.
+Overriding subscriptions only take effect during the lifecycle of the current cmdlet. It does not change the subscription in the context, and does not affect subsequent cmdlets.";
```

Comment:
Cannot use resource string because when VS regenerates `Resources.Designer.cs` all public becomes internal, but actually the resource strings are used by az.accounts, and nobody want to manually edit that file, so

Reply:
Actually Resources.resx allows us to specify default access modifier, not sure if this would help.